Sources/Zip/Zip.swift

```diff
@@ -92,40 +89,44 @@ public class Zip {
             currentPosition += Double(fileInfo.compressed_size)
             let fileNameSize = Int(fileInfo.size_filename) + 1
             let fileName = UnsafeMutablePointer<CChar>.allocate(capacity: fileNameSize)
+            defer { fileName.deallocate() }
 
             unzGetCurrentFileInfo64(zip, &fileInfo, fileName, UInt16(fileNameSize), nil, 0, nil, 0)
             fileName[Int(fileInfo.size_filename)] = 0
 
             var pathString = String(cString: fileName)
-            guard pathString.count > 0 else {
+
+            #if os(Windows)
+            // Colons are not allowed in Windows file names.
+            pathString = pathString.replacingOccurrences(of: ":", with: "_")
+            #endif
+
+            guard !pathString.isEmpty else {
                 throw ZipError.unzipFail
             }
 
             var isDirectory = false
-            let fileInfoSizeFileName = Int(fileInfo.size_filename-1)
+            let fileInfoSizeFileName = Int(fileInfo.size_filename - 1)
             if (fileName[fileInfoSizeFileName] == "/".cString(using: String.Encoding.utf8)?.first || fileName[fileInfoSizeFileName] == "\\".cString(using: String.Encoding.utf8)?.first) {
-                isDirectory = true;
+                isDirectory = true
             }
-            free(fileName)
             if pathString.rangeOfCharacter(from: CharacterSet(charactersIn: "/\\")) != nil {
                 pathString = pathString.replacingOccurrences(of: "\\", with: "/")
             }
 
-            let fullPath = destination.appendingPathComponent(pathString).standardized.path
-            // `.standardized` removes any `..` to move a level up.
+            let fullPath = destination.appendingPathComponent(pathString).standardizedFileURL.path
+            // `.standardizedFileURL` removes any `..` to move a level up.
             // If we then check that the `fullPath` starts with the destination directory we know we are not extracting "outside" the destination.
-            guard fullPath.starts(with: destination.standardized.path) else {
+            guard fullPath.starts(with: destination.standardizedFileURL.path) else {
                 throw ZipError.unzipFail
             }
 
-            let creationDate = Date()
             let directoryAttributes: [FileAttributeKey: Any]?
-            #if os(Linux) && swift(<6.0)
-                // On Linux, setting attributes is not yet really implemented.
-                // In Swift 4.2, the only settable attribute is `.posixPermissions`.
-                // See https://github.com/apple/swift-corelibs-foundation/blob/swift-4.2-branch/Foundation/FileManager.swift#L182-L196
+            #if (os(Linux) || os(Windows)) && swift(<6.0)
+                // On Linux and Windows, setting attributes is not yet really implemented.
```

Comment:
I'm not an expert but is this a problem? If so we should at least add a warning for people not to expect this to work

Reply:
I'm not an expert either, it was always like that in the original repo. In Swift 6 they can be set on all platforms, but looking at [the docs](https://developer.apple.com/documentation/foundation/fileattributekey/1418187-creationdate) right now I discovered that we may need to add a privacy manifest if we set any attribute outside of `.posixPermissions`

Reply:
I've decided to just remove the `.creationDate` and `.modificationDate` attributes on all platforms. Doesn't seem to be a problem, ZIPFoundation doesn't use them either

Reply:
ZIPFoundation sets them only on Apple platforms, now they are available in `swift-foundation` on all platforms so for Swift 6.0+ we set them on Linux and Windows too. Doesn't seem to be a problem. Regarding the privacy manifest, this a package oriented more on the server-side, but if anyone wants to open a PR to add it we can add it